src/main/java/com/example/auctionapp/user/UserService.java

```diff
@@ -0,0 +1,32 @@
+package com.example.auctionapp.user;
+
+import lombok.AllArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class UserService implements UserDetailsService {
+
+    private UserRepository userRepository;
+    private final BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    @Override
+    public UserDetails loadUserByUsername(String s) throws UsernameNotFoundException {
+        return userRepository.findByEmail(s).orElseThrow(() -> new UsernameNotFoundException("User with that email does not exist"));
+    }
+
+    public String signUpUser(User user){
+        boolean userExists = userRepository.findByEmail(user.getEmail()).isPresent();
+        if(userExists){
+            return("User with this email already exists");
```

Comment:
Extract this string as const `private static final String USER_EXISTS_RESPONSE = "User with this email already exists"`. So this becomes return(USER_EXISTS_RESPONSE);
Do same for success response